qutip/qip/circuit.py

```diff
@@ -383,9 +399,8 @@ def add_circuit(self, qc, start=0):
                               gate.arg_label)
             elif gate.name in ["BERKELEY", "SWAPalpha", "SWAP", "ISWAP",
```

Comment:
Maybe here we can also use the nice `_swap_like` category you defined?

Reply:
Sure. I'm working on modify the if/else here. I just found more bugs, like the `SWAPalpha` is a parameterized gate, but its parameter will be set to `None` after `add_circuit`. I need to double check if there are more bugs.

But another thing worries me is that, do we really need these if/else? The parameter of an unparameterized gate is already `None`. It looses nothing to pass `None` to the new gate. What we really need is to check if `targets` or `controls` are `None` right? The type of gate doesn't matter here.

Reply:
Indeed, it could be that we don't need those if/else at all. It seems to me that we can `deepcopy` the Gate and then modify only `targets` and `controls`. Maybe I'm missing something.